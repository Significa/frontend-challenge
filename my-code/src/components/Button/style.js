import sc, { css } from 'styled-components'
import { theme } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.button`
  display: flex;
  align-items: center;
  /* Account for border */
  padding: 0 15px 0 11px;
  height: 40px;
  outline: none;
  background: none;
  color: ${theme('greyLight')};
  border: 1px solid ${theme('greyMid')};
  border-radius: 4px;
  cursor: pointer;
  transition-timing-function: ${theme('transitionTiming')};
  transition-duration: ${theme('transitionDuration')};
  transition-property: color, border-color, background-color;
  font-family: inherit;
  ${text4}
  font-weight: 500;

  &:focus,
  &:hover {
    border-color: ${theme('mainRed')};
    color: ${theme('greyWhite')};
  }

  ${({ active }) =>
    active &&
    css`
      border-color: ${theme('mainRed')};
      background-color: ${theme('mainRed')};
      color: ${theme('greyWhite')};
    `}
`

export const IconWrapper = sc.div`
  width: 16px;
  height: 16px;
  margin-right: 12px;
  display: flex;
  align-items: center;
`
