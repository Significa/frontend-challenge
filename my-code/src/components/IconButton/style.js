import sc from 'styled-components'
import { theme } from 'styled-tools'

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = sc.button`
  height: 40px;
  outline: none;
  border: none;
  padding: 0;
  background: none;
  color: ${theme('greyLight')};
  cursor: pointer;
  transition:
    color
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  display: flex;
  align-items: center;
  justify-content: center;

  &:focus,
  &:hover {
    color: ${theme('greyWhite')};
  }
`
