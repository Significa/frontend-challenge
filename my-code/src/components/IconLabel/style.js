import sc from 'styled-components'
import { theme, prop } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.div`
  display: flex;
  align-items: stretch;
  height: 40px;
  margin-right: 16px;
  margin-bottom: 8px;
`

export const IconWrapper = sc.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${prop('color')};
  border-radius: 4px 0 0 4px;
  padding: 8px;
  min-width: 40px;
  box-sizing: border-box;
`

export const TextWrapper = sc.div`
  border: 1px solid ${theme('greyMid')};
  border-left: none;
  /* Account for border */
  padding: 8px 7px 8px 8px;
  border-radius: 0 4px 4px 0;
  ${text4}
  font-weight: 500;
`
