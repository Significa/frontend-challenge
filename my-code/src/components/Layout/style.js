import sc from 'styled-components'

// eslint-disable-next-line import/prefer-default-export
export const Outer = sc.div`
  display: flex;
  flex-direction: column;
  max-width: 1228px;
  margin: 0 auto;
  padding: 24px;
  min-height: 100vh;
  box-sizing: border-box;
`
