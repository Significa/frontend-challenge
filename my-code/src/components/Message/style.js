import sc from 'styled-components'
import { theme } from 'styled-tools'

import { text2, text4 } from '../../styles/text'

import illustration from '../../assets/illustration-empty-state@2x.png'

export const Center = sc.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`

export const Illustration = sc.img.attrs({
  src: illustration,
  alt: ''
})`
  object-fit: contain;
  width: 400px;
  max-width: 100%;
  height: 200px;
`

export const Title = sc.span`
  ${text2}
  margin-top: 16px;
`

export const Subtitle = sc.span`
  ${text4}
  color: ${theme('greyLight')};
  margin-top: 8px;
`
