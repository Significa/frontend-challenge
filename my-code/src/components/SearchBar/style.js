import sc from 'styled-components'
import { theme } from 'styled-tools'
import placeholder from 'polished/lib/mixins/placeholder'

import { text4 } from '../../styles/text'
import Magnifier from '../../assets/icon-magnifier.svg'

export const Wrapper = sc.div`
  margin: 24px 0;
  padding-left: 12px;
  border-radius: 4px;
  color: ${theme('greyDark')};
  background: ${theme('greyWhite')};
  display: flex;
  align-items: center;
`

export const Icon = sc(Magnifier)`
  color: ${theme('greyLight')};
`

export const Input = sc.input`
  background: none;
  border: none;
  margin: 0;
  font: inherit;
  ${text4}
  color: ${theme('greyDark')};
  display: block;
  padding: 12px;
  height: 44px;
  flex: 1;
  box-sizing: border-box;
  overflow: hidden;
  text-overflow: ellipsis;

  &:focus {
    outline: none;
  }

  ${placeholder({ color: theme('greyLight') })}
`
