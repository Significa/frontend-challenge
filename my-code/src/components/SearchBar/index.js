import React from 'react'
import PropTypes from 'prop-types'

import { Wrapper, Icon, Input } from './style'

export default function SearchBar({ value, onChange }) {
  return (
    <Wrapper>
      <Icon />
      <Input
        value={value}
        placeholder="Search movies..."
        onChange={event => onChange(event.target.value)}
      />
    </Wrapper>
  )
}

SearchBar.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}
