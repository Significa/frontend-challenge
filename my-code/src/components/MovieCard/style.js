import sc from 'styled-components'
import { Link } from 'react-router-dom'
import { theme } from 'styled-tools'

import { text3, text4 } from '../../styles/text'
import IconButton from '../IconButton'

export const Wrapper = sc.div`
  width: 100%;
  text-decoration: none;
  position: relative;
  border-radius: 3px;
  overflow: hidden;
  color: ${theme('greyWhite')};

  &:focus {
    outline: none;
  }

  /* Aspect ratio */
  &:before {
    content: '';
    display: block;
    padding-top: 133.3333333%;
  }
`

export const CoverLink = sc(Link)`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

export const Poster = sc.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: ${theme('greyMain')};
  background-image: ${({ poster }) => `url(${poster})`};
`

export const FavoriteButtonWrapper = sc(IconButton)`
  height: auto;
  position: absolute;
  z-index: 1;
  top: 12px;
  right: 12px;
  color: ${theme('greyWhite')};
  will-change: opacity;
  transition:
    opacity
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  filter: drop-shadow(0 0 2px rgba(0, 0, 0, .2));
  opacity: ${({ visible = true }) => (visible ? 1 : 0)};

  ${Poster} ~ ${CoverLink}:hover ~ &,
  ${Poster} ~ ${CoverLink}:focus ~ &,
  ${Poster} ~ &:hover,
  ${Poster} ~ &:focus {
    opacity: 1;
  }
`

export const Title = sc.span`
  ${text3}
`

export const Subtitle = sc.span`
  ${text4}
`

export const Content = sc.div`
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  padding: 12px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background: ${theme('scrimColor')};
  will-change: opacity;
  transition:
    opacity
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;

  ${Poster} ~ & {
    opacity: 0;
  }

  ${Poster} ~ ${CoverLink}:hover ~ &,
  ${Poster} ~ ${CoverLink}:focus ~ &,
  ${Poster} ~ ${FavoriteButtonWrapper}:hover ~ &,
  ${Poster} ~ ${FavoriteButtonWrapper}:focus ~ & {
    opacity: 1;
  }
`
