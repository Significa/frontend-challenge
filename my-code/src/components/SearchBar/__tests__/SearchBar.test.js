import React from 'react'
import { shallow } from 'enzyme'

import SearchBar from '../'
import { Wrapper, Icon, Input } from '../style'

describe('SearchBar', () => {
  it('renders correctly', () => {
    const tree = shallow(<SearchBar value="search value" onChange={() => {}} />)
    expect(tree).toMatchSnapshot()
  })

  it('calls onChange when values changes', () => {
    const mockCallback = jest.fn()
    const tree = shallow(<SearchBar value="" onChange={mockCallback} />)
    tree.find(Input).simulate('change', { target: { value: 'new value' } })
    expect(mockCallback.mock.calls.length).toBe(1)
    expect(mockCallback.mock.calls[0][0]).toBe('new value')
  })

  describe('Wrapper', () => {
    it('renders correctly', () => {
      const tree = shallow(<Wrapper />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Icon', () => {
    it('renders correctly', () => {
      const tree = shallow(<Icon />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Input', () => {
    it('renders correctly', () => {
      const tree = shallow(<Input />)
      expect(tree).toMatchSnapshot()
    })
  })
})
