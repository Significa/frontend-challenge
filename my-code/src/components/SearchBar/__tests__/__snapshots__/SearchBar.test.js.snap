// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Icon renders correctly 1`] = `
.c0 {
  color: theme.greyLight;
}

<Component
  className="c0"
/>
`;

exports[`SearchBar Input renders correctly 1`] = `
.c0 {
  background: none;
  border: none;
  margin: 0;
  font: inherit;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  color: theme.greyDark;
  display: block;
  padding: 12px;
  height: 44px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0:focus {
  outline: none;
}

.c0::-webkit-input-placeholder {
  color: theme.greyLight;
}

.c0:-moz-placeholder {
  color: theme.greyLight;
}

.c0::-moz-placeholder {
  color: theme.greyLight;
}

.c0:-ms-input-placeholder {
  color: theme.greyLight;
}

<input
  className="c0"
/>
`;

exports[`SearchBar Wrapper renders correctly 1`] = `
.c0 {
  margin: 24px 0;
  padding-left: 12px;
  border-radius: 4px;
  color: theme.greyDark;
  background: theme.greyWhite;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`SearchBar renders correctly 1`] = `
<Wrapper>
  <Icon />
  <Input
    onChange={[Function]}
    placeholder="Search movies..."
    value="search value"
  />
</Wrapper>
`;
